### ISSUE TITLE
Requirements for packages for auto-update

### ISSUE THREAD

**Issue by kolloch** [ryantm/nixpkgs-update#198]
It would be awesome to document the requirements for making a package auto-updateable. At least, it isn't clear to me.

For context, see e.g. this comment and the following:

https://github.com/kolloch/crate2nix/issues/102#issuecomment-623959917

**Comment by ryantm** [comment]
Good question, and this should get better documented, but for now, I'll try to address this in this comment:

1. The source URL must dynamically update, if the version is updated
2. There can only be one source
3. If there are srcs for different systems, it will only update x86-64-linux
4. It can't have extra deps files like gemset.nix (ruby) or deps.nix (go)
5. It can't be one of the [blacklisted packages](https://github.com/ryantm/nixpkgs-update/blob/master/src/Blacklist.hs)

**Comment by amesgen** [comment]
> It can't have extra deps files like gemset.nix (ruby) or deps.nix (go)

In principle, would you be interested in/open to relaxing this?

**Comment by zowoq** [comment]
> > It can't have extra deps files like gemset.nix (ruby) or deps.nix (go)
> 
> In principle, would you be interested in/open to relaxing this?

This is already supported, packages can use [nix-update](https://github.com/Mic92/nix-update) (which can handle several of those types of files) as an updateScript or they set their own custom updateScript.

**Comment by amesgen** [comment]
Yes, but isn't the problem that nixpkgs-update doesn't find the old version in the main `.nix` file, and hence won't proceed in opening a PR?

For example, for `cddl`, which has a standard Ruby update script, but isn't updated by nixpkgs-update for this reason AFAICT, see
```
Old version 0.10.3" not present in master derivation file with contents: ...
```
in https://nixpkgs-update-logs.nix-community.org/cddl/2025-01-19.log
